ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -672,6 +674,7 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
 
         let chainDiffs = NE.nonEmpty
               $ map Diff.extend
+              $ filter (followsLoEFrag loeFrag)
```

Comment:
Don't we have to do sth similar in `switchToAFork`?

Reply:
I think you are right. I amended that.